examples/miscellaneous/miscellaneous_ex12/miscellaneous_ex12.C

```diff
@@ -0,0 +1,751 @@
+  for (unsigned int nid=1; nid<mesh.n_nodes(); ++nid)
+    {
+      const Real dist_sq = (mesh.point(nid)-point_D).size_sq();
+      if (dist_sq < nearest_dist_sq)
+        {
+          nearest_dist_sq = dist_sq;
+          node_D = mesh.node_ptr(nid);
+        }
+    }
+
+  // Evaluate the y-displacement "v" at the point D.
+  const unsigned int v_var = system.variable_number ("v");
+  dof_id_type v_dof = node_D->dof_number (system.number(), v_var, 0);
+  Number v = 0;
+  if (v_dof >= system.get_dof_map().first_dof() &&
+      v_dof <  system.get_dof_map().end_dof())
+    v = system.current_solution(v_dof);
+  system.comm().sum(v);
+
+  Real v_D_bar = E*h*v/q ;
+  const Real v_D_bar_analytic = 4.114;
+
+  // Print the finite element solution and the analytic
+  // prediction to the screen.
+  libMesh::out << "y-displacement of the point D: " << v_D_bar << std::endl;
+  libMesh::out << "Analytic solution: " << v_D_bar_analytic << std::endl;
+
+  // All done.
+  return 0;
+}
+
+// We now define the matrix and rhs vector assembly function
+// for the shell system.
+void assemble_shell (EquationSystems & es,
+                     const std::string & system_name)
+{
+  // It is a good idea to make sure we are assembling
+  // the proper system.
+  libmesh_assert_equal_to (system_name, "Shell");
+
+  // Get a constant reference to the mesh object.
+  const MeshBase & mesh = es.get_mesh();
+  const unsigned int dim = mesh.mesh_dimension();
+
+  // Get a reference to the shell system object.
+  LinearImplicitSystem & system = es.get_system<LinearImplicitSystem> ("Shell");
+
+  // Get the shell parameters that we need during assembly.
+  const Real h  = es.parameters.get<Real> ("thickness");
+  const Real E  = es.parameters.get<Real> ("young's modulus");
+  const Real nu = es.parameters.get<Real> ("poisson ratio");
+  const Real q  = es.parameters.get<Real> ("point load");
+
+  // The membrane elastic matrix.
+  Eigen::Matrix3d Hm;
+  Hm <<
+    1.,nu,0.,
+    nu,1.,0.,
+    0.,0.,0.5 * (1-nu);
+  Hm *= h * E/(1-nu*nu);
+
+  // The bending elastic matrix.
+  Eigen::Matrix3d Hf;
+  Hf <<
+    1.,nu,0.,
+    nu,1.,0.,
+    0.,0.,0.5 * (1-nu);
+  Hf *= h*h*h/12 * E/(1-nu*nu);
+
+  // The shear elastic matrices.
+  Eigen::Matrix2d Hc0 = Eigen::Matrix2d::Identity() ;
+  Hc0 *= h * 5./6*E/(2*(1+nu));
+
+  Eigen::Matrix2d Hc1 = Eigen::Matrix2d::Identity() ;
+  Hc1 *= h*h*h/12 * 5./6*E/(2*(1+nu));
+
+  // Get the Finite Element type, this will be
+  // the same for all variables.
+  FEType fe_type = system.variable_type (0);
+
+  UniquePtr<FEBase> fe (FEBase::build(dim, fe_type));
+  QGauss qrule (dim, fe_type.default_quadrature_order());
+  fe->attach_quadrature_rule (&qrule);
+
+  // The element Jacobian * quadrature weight at each integration point.
+  const std::vector<Real> & JxW = fe->get_JxW();
+
+  // The element shape function and its derivatives evaluated at the
+  // quadrature points.
+  const auto & dxyzdxi = fe->get_dxyzdxi();
```

Comment:
Sorry, still no C++11 code allowed in examples for the time being.  (I will fix any small stuff like this, no need to make extra commits.)